app/Models/Campaign.php

```diff
@@ -303,6 +303,12 @@ public function toSearchableArray()
             $array[$date] = optional($this->{$date})->timestamp;
         }
 
+        // If a campaign doesn't have an end date, we'll set a far-future one in
+        // Algolia to allow filtering. More context: <...>
+        if (empty($array['end_date'])) {
+            $array['end_date'] = 2147483647; // January 19th 2038
+        }
```

Comment:
This was a surprise to me: because of [restrictions on combining ANDs and ORs](https://www.algolia.com/doc/guides/managing-results/refine-results/filtering/in-depth/combining-boolean-operators/#combining-ands-and-ors), we can't simply say "find me campaigns where the start date is in the past, and the end_date exists & is in the future".

To allow us to query for "open" campaigns, I add a far-future "end date" here whenever the `end_date` field is null, so that we can [make a simple date comparison query for Algolia](https://github.com/DoSomething/graphql/pull/247/files#diff-a8806c93ceaf69be83f446966bbf9974R903-R904). 

This "January 19th 2038" date is just the date version of the maximum 32-bit integer (which, as far as I understand, is also the maximum date that we can safely encode into a UNIX timestamp). And, as much as it pains me to say, I'm guessing that we'll probably re-write this application _sometime_ befro 2038 rolls around. 😉

Reply:
The terminators will take care of it!